spring-cloud-gcp-config/src/main/java/org/springframework/cloud/gcp/config/GoogleConfigPropertySourceLocator.java

```diff
@@ -0,0 +1,171 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.config;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.ServiceOptions;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cloud.bootstrap.config.PropertySourceLocator;
+import org.springframework.cloud.config.client.ConfigClientProperties;
+import org.springframework.core.annotation.Order;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Custom property source locator for Google Cloud Runtime Configurator.
+ *
+ * @author Jisha Abubaker
+ */
+@Order(0)
+public final class GoogleConfigPropertySourceLocator implements PropertySourceLocator {
+
+	private static final String RUNTIMECONFIG_API_ROOT = "https://runtimeconfig.googleapis.com/v1beta1/";
+
+	private static final String RUNTIMECONFIG_SCOPE = "https://www.googleapis.com/auth/cloudruntimeconfig";
+
+	private static final String ALL_VARIABLES_PATH =
+			"projects/{project}/configs/{name}_{profile}/variables?returnValues=true";
+
+	private static final String PROPERTY_SOURCE_NAME = "spring-cloud-gcp";
+
+	private static final String AUTHORIZATION_HEADER = "Authorization";
+
+	private static final String BEARER_TOKEN_PREFIX = "Bearer ";
+
+	private static final Log logger = LogFactory.getLog(GoogleConfigPropertySourceLocator.class);
+
+	private ConfigClientProperties defaultProperties;
+
+	private GoogleCredentials credentials;
+
+	private String projectId;
+
+	public GoogleConfigPropertySourceLocator(ConfigClientProperties defaultProperties) throws IOException {
+		this.defaultProperties = defaultProperties;
+		this.credentials = ServiceAccountCredentials.getApplicationDefault()
```

Comment:
Should we instead add a `CredentialsProvider` to the ctor args and use that? In order to use the credentials specified by the user.